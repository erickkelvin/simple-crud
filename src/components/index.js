import List from './List/List.js';
import ListItem from './ListItem/ListItem.js';
import Form from './Form/Form.js';
import Field from './Field/Field.js';
import Button from './Button/Button.js';
import Spinner from './Spinner/Spinner.js';
import NotFound from './NotFound/NotFound.js';

export {
  List,
  ListItem,
  Form,
  Field,
  Button,
  Spinner,
  NotFound
};