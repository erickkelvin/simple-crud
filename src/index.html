<!DOCTYPE html>
<html>
  <head>
    <title>Simple CRUD</title>
    <script type="module" src="app.bundle.js"></script>
  </head>
  <body>
    <div id="root"></div>
    <div id="credits">Desenvolvido por Erick Kelvin</div>
  </body>
</html>