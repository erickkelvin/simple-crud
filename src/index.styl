html, body
  height: 100%

body
  display: flex
  flex-direction: column
  align-items: center
  justify-content: center
  background: #00c8b3
  font-family: sans-serif

#root
  background: #fff
  padding: 20px
  border-radius: 5px

#credits
  margin-top: 5px
  font-size: 10px
  color: #fff

.pageContainer
  animation: .5s showPage

@keyframes showPage
  0%
    opacity: 0
  100%
    opacity: 1

h3
  display: inline-block
  margin-top: 5px