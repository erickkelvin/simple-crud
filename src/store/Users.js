export default class Users {
  
  constructor() {
    this.users = {};
  }

  fetchInitialState(successCallback, errorCallback) {
    if (localStorage.getItem('users') === null) {
      fetch('https://private-21e8de-rafaellucio.apiary-mock.com/users')
      .then((response) => {
        if (response.ok) {
          response.json().then((users) => {
            users.forEach((userData) => {
              this.createUser(userData);
            });
            if (successCallback) {
              successCallback(this.users);
            }
          });
        } else {
          if (errorCallback) {
            errorCallback();
          }
          console.log('Could not fetch initial state.');
        }
      })
      .catch((error) => {
        if (errorCallback) {
          errorCallback();
        }
        console.log('Could not fetch initial state: ' + error.message);
      });
    } else {
      if (successCallback) {
        successCallback(this.getUsers());
      }
    }
  }

  getUsers() {
    this.users = JSON.parse(localStorage.getItem('users')) || this.users;
    return this.users;
  }

  setUsers() {
    localStorage.setItem('users', JSON.stringify(this.users));
  }

  getUser(userId) {
    this.getUsers();
    return this.users[userId];
  }

  createUser(userData) {
    this.getUsers();
    userData.id = Math.random().toString(36).substr(2, 9);
    userData.cpf = userData.cpf.replace(/^(\d{3})(\d{3})(\d{3})(\d{0,1})/,"$1.$2.$3-$4");
    userData.phone = userData.phone.replace(/^(\d{2})(\d{5})(\d{0,4}).*/,"($1) $2-$3");
    this.users[userData.id] = userData;
    this.setUsers();
  }

  updateUser(userId, userData) {
    this.getUsers();
    userData.id = userId;
    this.users[userId] = userData;
    this.setUsers();
  }

  deleteUser(userId) {
    this.getUsers();
    delete this.users[userId];
    this.setUsers();
  }
}