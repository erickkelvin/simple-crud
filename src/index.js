import { Form, List, NotFound } from 'components';
import { Users } from 'store';
import './index.styl';

const routes = {
  'index': List,
  'new': Form,
  'edit': Form,
  '404': NotFound
};

const store = new Users();

const renderApp = () => {
  if (window.location.pathname !== '/') {
    window.location.href = '/';
  }
  const path = window.location.hash.split('/');
  const route = routes[path[1] || 'index']  || routes['404'];
  const page = new route(store, path[1] === 'edit' && path[2]);
  document.getElementById('root').innerHTML = page.render();
  page.postRender && page.postRender();
};

renderApp();

window.addEventListener("hashchange", renderApp);
