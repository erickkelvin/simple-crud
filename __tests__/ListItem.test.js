import { List, ListItem } from 'components';
import { Users } from '../src/store';

const data = {
  id: 'id123',
  name: 'name',
  email: 'email@test.com',
  cpf: '123.456.789-12',
  phone: '(12) 34567-8901'
};

describe('ListItem test', () => {
  const listItem = new ListItem(new Users(), new List(), data);

  it('render', () => {
    expect(listItem.render()).toMatchSnapshot();
  });
});