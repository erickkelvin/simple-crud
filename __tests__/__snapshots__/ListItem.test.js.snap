// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem test render 1`] = `
"
      <div id='userid123' class='listItem'>
        <div class='listItemContent'>
          <div>
            <strong>Nome:</strong> name
          </div>
          <div>
            <strong>E-mail:</strong> email@test.com
          </div>
          <div>
            <strong>CPF:</strong> 123.456.789-12
          </div>
          <div>
            <strong>Telefone:</strong> (12) 34567-8901
          </div>
        </div>
        <div class='listItemActions'>
          
      <button id='editid123Button'>Editar</button>
    
          
      <button id='deleteid123Button'>Excluir</button>
    
        </div>
        
      </div>
    "
`;
